project3/participant/tasks/multicast_receiver_task.cpp

```diff
@@ -46,4 +44,10 @@ thread_pool::Task::Status MulticastReceiver::RunAtomic() {
 
 void MulticastReceiver::CleanUp() { close(messenger_fd_); }
 
+void MulticastReceiver::WriteLineToLog(const std::string& message) {
+  std::ofstream log_file;
```

Comment:
Might be worth just opening the log file once in `SetUp` and keeping the handle around.